src/main/scala/shine/DPIA/FunctionalPrimitives/Array.scala

```diff
@@ -14,9 +15,9 @@ final case class Array(dt: DataType,
   extends ExpPrimitive {
 
   private def tRec(m: Int): ExpType =
-    if (m <= 0) { exp"[${elements.length : Nat}.$dt, $read]" }
-    else { dt ->: tRec(m - 1) }
-  override val t: ExpType = (dt: DataType) ->: tRec(elements.length)
+    if (m <= 0) { expT({elements.length : Nat}`.`dt, read) }
+    else { tRec(m - 1) }
```

Comment:
I think that I can entirely remove the recursion and write the final array type `expT(elements.length`.`dt, read)` here.